bip-0340/reference.py

```diff
@@ -73,36 +76,40 @@ def lift_x_square_y(b):
         return None
     return (x, y)
 
-def lift_x_even_y(b):
+def lift_x_even_y(b: bytes) -> Optional[Point]:
     P = lift_x_square_y(b)
     if P is None:
         return None
     else:
         return (x(P), y(P) if y(P) % 2 == 0 else p - y(P))
 
-def int_from_bytes(b):
+def int_from_bytes(b: bytes) -> int:
     return int.from_bytes(b, byteorder="big")
 
-def hash_sha256(b):
+def hash_sha256(b: bytes) -> bytes:
     return hashlib.sha256(b).digest()
 
-def is_square(x):
-    return pow(x, (p - 1) // 2, p) == 1
+def is_square(x: int) -> bool:
+    return int(pow(x, (p - 1) // 2, p)) == 1
```

Comment:
What is this int conversion for? Does `pow` return another type?

Reply:
`pow` can return float. For example, try `pow(2, -1)`. This is also noted in the [common issues section of the Mypy manual](https://mypy.readthedocs.io/en/stable/common_issues.html). More discussion at [mypy issue 285](https://github.com/python/typeshed/issues/285). If you prefer to avoid the coercion (which won't actually coerce), I think it would be possible to use [cast](https://mypy.readthedocs.io/en/stable/casts.html), which is always guaranteed to be the identity function at runtime. Which do you prefer?

Reply:
Let's keep it like this. I suspect cast is more obscure to most readers.